### ISSUE TITLE
Not able to connect to ES Cluster because of blank ES Cluster name and HOST properties

### ISSUE THREAD

**Issue by preetpuri** [fhussonnois/storm-trident-elasticsearch#8]
In the Elastic Search ClientFactory.java makeClient() (under static class Transport) method is not getting elastic search properties from settings map instead it tries to get HOST and CLUSTER name from storm configuration map (in which these are not present). 

 @Override
        public TransportClient makeClient(Map conf) {
            String clusterHosts = (String)settings.get(HOSTS); // instead of conf.get(HOSTS);
            String clusterName  = (String)settings.get(NAME); // instead of conf.get(NAME);
            Preconditions.checkNotNull(clusterHosts,"no setting found for Transport Client, make sure that you set property " + HOSTS);
........................

**Comment by fhussonnois** [comment]
Hi, 
I am confused because everything seems correct into the class : [ClientFactory.java#L68](https://github.com/fhussonnois/storm-trident-elasticsearch/blob/master/src/main/java/com/github/fhuss/storm/elasticsearch/ClientFactory.java#L68)

**Comment by preetpuri** [comment]
The settings we are passing to ClientFactoty.Transport (see below) method has all the ES related configuration but in the makeClient method of Transport class we are picking ES host and name properties from different configuration map (which contains only storm related configurations not the one which we set)??

 Settings settings = ImmutableSettings.settingsBuilder().put("storm.elasticsearch.hosts", "localhost").put("storm.elasticsearch.cluster.name", "testingcluster").build();

StateFactory stateFactory = ESIndexMapState.nonTransactional(new ClientFactory.Transport(settings.getAsMap()), Map.class);

**Comment by fhussonnois** [comment]
Actually, properties should be set into the storm configuration : 

```
StormTopology topology = ...
LocalCluster cluster = new LocalCluster();
Config config = new Config();
config.put("storm.elasticsearch.cluster.name", "my-cluster");
config.put("storm.elasticsearch.hosts", "localhost");
cluster.submitTopology("elastic-storm", new Config(), topology);
```

Then, you can instantiate your StateFactory as follows : 

```
StateFactory stateFactory = ESIndexMapState.nonTransactional(new ClientFactory.Transport(), Map.class);
```

**Comment by preetpuri** [comment]
Yes it can be done that way but then what kind of settings should we set in Transport class?
public Transport(Map<String, String> settings) {
            this.settings = settings;
 }

**Comment by lester015** [comment]
why the bulk index not work?

I using this IndexBatchBolt class which can realize bulk indexs into elasticsearch,the storm submit the topology was successfully, the storm UI as flows

![image](https://cloud.githubusercontent.com/assets/12822989/21413076/cf1b11b0-c82f-11e6-906b-a8e5cede416e.png)

I see the  work log and found the bulkUpdateIndexes method not finnied. the work-port.log as bellows:
2016-12-22T10:18:15.711+0800 c.g.f.s.e.b.IndexBatchBolt [INFO] executeTickTuple is beginning ....source: __system:-1, stream: __tick, id: {}, [1]
2016-12-22T10:18:15.711+0800 c.g.f.s.e.b.IndexBatchBolt [INFO] bulkUpdateIndexes is beginning ....
2016-12-22T10:18:15.711+0800 c.g.f.s.e.b.IndexBatchBolt [INFO] queue size ....0
2016-12-22T10:18:15.711+0800 c.g.f.s.e.b.IndexBatchBolt [INFO]  inputs.size ...0
2016-12-22T10:18:16.710+0800 c.g.f.s.e.b.IndexBatchBolt [INFO] executeTickTuple is beginning ....source: __system:-1, stream: __tick, id: {}, [1]
2016-12-22T10:18:16.710+0800 c.g.f.s.e.b.IndexBatchBolt [INFO] bulkUpdateIndexes is beginning ....
2016-12-22T10:18:16.711+0800 c.g.f.s.e.b.IndexBatchBolt [INFO] queue size ....0
2016-12-22T10:18:16.711+0800 c.g.f.s.e.b.IndexBatchBolt [INFO]  inputs.size ...0
2016-12-22T10:18:16.711+0800 c.g.f.s.e.b.IndexBatchBolt [INFO] executeTickTuple is beginning ....source: __system:-1, stream: __tick, id: {}, [1]
2016-12-22T10:18:16.711+0800 c.g.f.s.e.b.IndexBatchBolt [INFO] bulkUpdateIndexes is beginning ....
2016-12-22T10:18:16.711+0800 c.g.f.s.e.b.IndexBatchBolt [INFO] queue size ....0
2016-12-22T10:18:16.711+0800 c.g.f.s.e.b.IndexBatchBolt [INFO]  inputs.size ...0

thanks,
